src/libsystemd-network/dhcp6-option.c

```diff
@@ -167,6 +167,33 @@ int dhcp6_option_append_fqdn(uint8_t **buf, size_t *buflen, const char *fqdn) {
         return r;
 }
 
+int dhcp6_option_append_user_class(uint8_t **buf, size_t *buflen, char **user_class) {
+        _cleanup_free_ uint8_t *p = NULL;
+        size_t total = 0, offset = 0;
+        char **s;
+
+        assert_return(buf && *buf && buflen && user_class, -EINVAL);
+
+        STRV_FOREACH(s, user_class) {
+                size_t len = strlen(*s);
+
+                if (len > 255)
+                        return -ENAMETOOLONG;
+
+                p = realloc(p, total + len + 2);
+                if (!p)
+                        return -ENOMEM;
+
+                unaligned_write_be16(&p[offset], len);
```

Comment:
hmm, so the size is a 16bit value, but only 0…255 are actually allowed? is that correct?

the snippet from the RFC you used as commit msg (great idea btw!) isn't clear about that... i.e. arent' lengths in the range 256…65535 OK too?

Reply:
Yes it's not clearly written. I guess would be OK . Updated.